src/Adapter/PlatformServices.WinUI/AppModel.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#nullable enable
+#pragma warning disable SA1310 // Field names must not contain underscore
+
+namespace Microsoft.VisualStudio.TestPlatform.MSTestAdapter.PlatformServices.AppContainer
+{
+    using System;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    /// <summary>
+    /// Global helpers for queriying package information
+    /// </summary>
+    internal static class AppModel
+    {
+        /// <summary>
+        /// Checks whether the current application is packaged or not.
+        /// </summary>
+        /// <returns>Returns true if application is packaged; false otherwise.</returns>
+        /// <see href="https://github.com/microsoft/WindowsAppSDK/blob/f1cd5fc8ce6a6fed8442b3f64978b9e65b1d43fd/dev/Common/AppModel.Identity.h#L11"/>
+        public static bool IsPackagedProcess()
+        {
+            int result, packageFullNameLength = 0;
+
+            try
+            {
+                result = Interop.GetCurrentPackageFullName(ref packageFullNameLength, null);
+
+                /* Second call will return the package name.
+                var packageFullName = new StringBuilder(packageFullNameLength);
+                result = GetCurrentPackageFullName(ref packageFullNameLength, packageFullName);
+                */
+            }
+            catch (EntryPointNotFoundException)
+            {
+                return false;
+            }
+
+            return result != Interop.APPMODEL_ERROR_NO_PACKAGE && packageFullNameLength != 0;
```

Comment:
I am not sure to understand why they also check `ERROR_INSUFFICIENT_BUFFER`, but is it something we should care about here?

Reply:
I don't know the API, but I assume if the package name is null - it's not a package :)